### PR TITLE
Add TS API bindings for PWA

### PR DESCRIPTION
**Pull request by tervay**
## Description
Adds TS bindings for fetching data from TBA utilizing [oazapfts](https://github.com/oazapfts/oazapfts). During my testing I found this library to be the most robust and flexible for generating TS OpenAPI clients.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change API specifications or require data migrations)

### CODE REVIEW

File: pwa/README.md [hunk 1]
```diff
@@ -4,6 +4,12 @@
 
 ## Development
 
+Make sure you have your TBA APIv3 Read Key set in `.env`:
+
+```sh
+VITE_TBA_API_READ_KEY="myKey"
```
Comment (fangeugene) [inline pwa/README.md:10]: I will work on getting this key set up with CI deployments

File: pwa/package.json [hunk 2]
```diff
@@ -10,9 +10,11 @@
     "format:check": "prettier --check .",
     "format:fix": "prettier --write .",
     "start": "remix-serve ./build/server/index.js",
-    "typecheck": "tsc"
+    "typecheck": "tsc",
+    "generate-api": "oazapfts ../src/backend/web/static/swagger/api_v3.json --optimistic --argumentStyle=object > app/api/v3.ts"
```
Comment (fangeugene) [inline pwa/package.json:14]: Does this need to get regenerated after #6071 landed?

File: pwa/package.json [hunk 3]
```diff
@@ -34,6 +36,7 @@
     "eslint-plugin-jsx-a11y": "^6.7.1",
     "eslint-plugin-react": "^7.33.2",
     "eslint-plugin-react-hooks": "^4.6.0",
+    "oazapfts": "^6.1.0",
```
Comment (fangeugene) [inline pwa/package.json:39]: I haven't used this in the past, but I've used `openapi-typescript`. Did you look into the differences between the two? I don't have strong opinions either way but `openapi-typescript` looks to be used more.

Comment (tervay) [reply pwa/package.json:39]: Yeah I tried out both. `openapi-typescript` is definitely more popular BUT for our specific API I think it's much more clunky:

```ts
const { data, error } = await client.GET("/team/{team_key}", {
  params: {
    path: { team_key: "frc604" },
  },
});
```

It's _really_ verbose. `oazapfts` is simply `await client.getTeam("frc604");`

Let me do one more round of digging into this to make sure I haven't missed anything with `openapi-typescript`.

Comment (tervay) [reply pwa/package.json:39]: Took another look and not seeing a way to do it with `openapi-typescript` that makes logical sense and is easy to use. So I think oazapfts is the way to go.

Comment (fangeugene) [reply pwa/package.json:39]: Fair enough. Can you fix the package.json conflicts then i can merge

File: pwa/package.json (hunk 2 above)
Comment (tervay) [reply pwa/package.json:14]: yes

Comment (tervay) [reply pwa/package.json:14]: done

**Pull request merged**